pkg/interfaces/http/handlers/api_handler.go

```diff
@@ -0,0 +1,38 @@
+package handlers
+
+import (
+	"fmt"
+	"github.com/3scale-labs/kamwiel/pkg/domain/api"
+	"github.com/gin-gonic/gin"
+	"net/http"
+)
+
+type APIHandler interface {
+	Get(*gin.Context)
+}
+
+type apiHandler struct {
+	service api.Service
+}
+
+func NewAPIHandler(service api.Service) APIHandler {
+	return &apiHandler{
+		service: service,
+	}
+}
+
+func (h *apiHandler) Get(ctx *gin.Context) {
+	name := ctx.Param("name")
+	if len(name) == 0 {
+		ctx.JSON(http.StatusBadRequest, "Missing param `name`")
+		return
+	}
+	api, getErr := h.service.GetAPI(name)
+	if getErr != nil {
+		fmt.Println("API not found", getErr)
```

Comment:
for future PR, when there is an error, it can be something else than not found. It can be just connectivity issues or whatever. There is helper function that tells you if it is a not found error. It is used [here](https://github.com/Kuadrant/kuadrant-controller/blob/093fb4676afbaa965ee6b5e9269eec6f31382db7/controllers/service_controller.go#L82)